quartz_solar_forecast/eval/forecast.py

```diff
@@ -0,0 +1,89 @@
+import os
+import pandas as pd
+import xarray as xr
+from psp.data_sources.nwp import NwpDataSource
+from psp.data_sources.pv import NetcdfPvDataSource
+from psp.serialization import load_model
+from psp.typings import X
+
+from quartz_solar_forecast.data import get_nwp, make_pv_data
+from quartz_solar_forecast.pydantic_models import PVSite
+from quartz_solar_forecast.forecasts.v1 import forecast_v1
+from quartz_solar_forecast.data import format_nwp_data
+
+from datetime import datetime
+
+dir_path = os.path.dirname(os.path.realpath(__file__))
+
+
+def run_forecast(pv_df: pd.DataFrame, nwp_df: pd.DataFrame, nwp_source="ICON") -> pd.DataFrame:
+    """
+    Run the forecast from NWP data
+
+    :param pv_df: the PV site data. This should have columns timestamp, id, latitude, longitude, and capacity
+    :param nwp_df: all the nwp data for the site and location. This shoulw have the following rows
+        - timestamp: the timestamp of the site
+        - temperature_2m
+        - precipitation
+        - shortwave_radiation
+        - direct_radiation",
+        - cloudcover_low",
+        - cloudcover_mid",
+        - cloudcover_high",
+        maybe more
+    """
+
+    # load model only once
+    model = load_model(f"{dir_path}/../models/model-0.3.0.pkl")
+
+    all_predictions = []
+    for i in range(len(pv_df)):
+
+        print(f"Running forecast for {i} of {len(pv_df)}")
+
+        pv_row = pv_df.iloc[i]
+
+        site = PVSite(
+            latitude=pv_row["latitude"],
+            longitude=pv_row["longitude"],
```

Comment:
Good next task, that i could do could be add in the tilt and orientation code

Reply:
#20

Reply:
Yes, we would need to.